core/trino-main/src/main/java/io/trino/sql/gen/LambdaBytecodeGenerator.java

```diff
@@ -194,16 +194,19 @@ public static BytecodeNode generateLambda(
         Variable wasNull = scope.getVariable("wasNull");
 
         // generate values to be captured
-        ImmutableList.Builder<BytecodeExpression> captureVariableBuilder = ImmutableList.builder();
+        ImmutableList.Builder<BytecodeExpression> captureVariableBuilder = ImmutableList.builderWithExpectedSize(captureExpressions.size());
+        ImmutableList.Builder<Variable> tempVariablesBuilder = ImmutableList.builderWithExpectedSize(captureExpressions.size());
```

Comment:
You could just use an array list here instead of a builder

Reply:
Done